src/category_theory/linear/default.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.preadditive
+import algebra.module.linear_map
+import algebra.invertible
+
+/-!
+# Linear categories
+
+A `R`-linear category is a category in which `X ⟶ Y` is an `R`-module in such a way that
+composition of morphisms is `R`-linear in both variables.
+
+## Implementation
+
+Corresponding to the fact that we need to have an `add_comm_group X` structure in place
+to talk about a `module R X` structure,
+we need `preadditive C` as a prerequisite typeclass for `linear R C`.
+This makes for longer signatures than would be ideal.
+
+## Future work
+
+It would be nice to have a usable framework of enriched categories in which this just became
+a category enriched in `Module R`.
+
+-/
+
+universes w v u
+
+open category_theory.limits
+open linear_map
+
+namespace category_theory
+
+/-- A category is called `R`-linear if `P ⟶ Q` is an `R`-module such that composition is
+    `R`-linear in both variables. -/
+class linear (R : Type w) [ring R] (C : Type u) [category.{v} C] [preadditive C] :=
+(hom_module : Π X Y : C, module R (X ⟶ Y) . tactic.apply_instance)
+(smul_comp' : ∀ (X Y Z : C) (r : R) (f : X ⟶ Y) (g : Y ⟶ Z),
+  (r • f) ≫ g = r • (f ≫ g) . obviously)
+(comp_smul' : ∀ (X Y Z : C) (f : X ⟶ Y) (r : R) (g : Y ⟶ Z),
+  f ≫ (r • g) = r • (f ≫ g) . obviously)
+
+attribute [instance] linear.hom_module
+restate_axiom linear.smul_comp'
+restate_axiom linear.comp_smul'
+attribute [simp,reassoc] linear.smul_comp
+attribute [reassoc] linear.comp_smul -- (the linter doesn't like `simp` on this lemma)
+attribute [simp] linear.comp_smul
+
+end category_theory
+
+open category_theory
+
+namespace category_theory.linear
+
+variables {R : Type w} [ring R] {C : Type u} [category.{v} C] [preadditive C] [linear R C]
+
+section induced_category
+universes u'
+variables {C} {D : Type u'} (F : D → C)
+
+instance induced_category.category : linear.{w v} R (induced_category C F) :=
+{ hom_module := λ X Y, @linear.hom_module R _ C _ _ _ (F X) (F Y),
+  smul_comp' := λ P Q R f f' g, smul_comp' _ _ _ _ _ _,
+  comp_smul' := λ P Q R f g g', comp_smul' _ _ _ _ _ _, }
+
+end induced_category
+
+variables (R)
+
+/-- Composition by a fixed left argument as an `R`-linear map. -/
+@[simps]
+def left_comp {X Y : C} (Z : C) (f : X ⟶ Y) : (Y ⟶ Z) →ₗ[R] (X ⟶ Z) :=
+{ to_fun := λ g, f ≫ g,
+  map_add' := by simp,
+  map_smul' := by simp, }
+
+/-- Composition by a fixed right argument as an `R`-linear map. -/
```

Comment:
Is it easier here to construct the `R`-linear map of composition in both arguments, like
```
(X ⟶ Y) →ₗ[R] ((Y ⟶ Z) →ₗ[R] (X ⟶ Z))
```
and make these two out of the bigger one? If not, perhaps having this as a consequence could be convenient.

Reply:
This requires the ring to be commutative, so I've done it lower down in the file.